pkg/util/acrtoken/acrtoken.go

```diff
@@ -0,0 +1,49 @@
+package acrtoken
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerregistry/mgmt/2019-06-01-preview/containerregistry"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/date"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+)
+
+// CreateACRToken create a token on our registry
+// see https://docs.microsoft.com/en-us/azure/container-registry/container-registry-repository-scoped-permissions
+func CreateACRToken(ctx context.Context, env env.Interface, tokenName, secret string) error {
+	authorizer, err := env.FPAuthorizer(env.TenantID(), azure.PublicCloud.ResourceManagerEndpoint)
```

Comment:
call this localFPAuthorizer for consistency
pull it out to simplify unit testing